### PR TITLE
NIFI-4948 - MongoDB Lookup Service throws an exception if there is no…

### PR DESCRIPTION
**Pull request by pvillard31**
… match

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by pvillard31** [comment]
@MikeThomsen - since you've been extensively working on the Mongo processors, I'd appreciate if you could give your feedback on this one. Thanks much!

**Comment by MikeThomsen** [comment]
@pvillard31 Sure, np.

File: nifi-nar-bundles/nifi-standard-services/nifi-mongodb-services-bundle/nifi-mongodb-services/src/main/java/org/apache/nifi/mongodb/MongoDBControllerService.java [hunk 1]
```diff
@@ -69,9 +70,10 @@ public boolean exists(Document query) {
         return this.col.count(query) > 0;
     }
 
+    @Override
     public Document findOne(Document query) {
         MongoCursor<Document> cursor  = this.col.find(query).limit(1).iterator();
-        Document retVal = cursor.next();
+        Document retVal = cursor.tryNext();
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-standard-services/nifi-mongodb-services-bundle/nifi-mongodb-services/src/main/java/org/apache/nifi/mongodb/MongoDBControllerService.java]: Ok, this looks right based on the [official documentation](http://api.mongodb.com/java/current/com/mongodb/client/MongoCursor.html#tryNext--)

**Comment by mattyb149** [comment]
+1 LGTM, thanks to all for the reviews, and to Pierre for the improvement! Merging to master

**Pull request closed without merging**